deploy/kubernetes/elastic-agent-kustomize/default/README.md

```diff
@@ -0,0 +1,71 @@
+# Kustomize Templates
+
+The list below includes the official [kustomize](https://github.com/kubernetes-sigs/kustomize) templates to run them in Kubernetes:
+
+Agent Scenario | Description
+---- | ----
+[Elastic Agent managed - Default ](./elastic-agent-managed/) | Default Elastic Agent managed by Fleet setup. Kube-state-metrics (KSM) is installed automatically.
+[Elastic Agent standalone Default ](./elastic-agent-standalone/) | Default Standalone Elastic Agent setup. Kube-state-metrics (KSM) is installed automatically.
+
+## Using above templates
+
+Users can clone this repository to use the provided kustomize templates.
+
+For *Managed Elastic Agent*, please update the following variables inside main kustomization.yaml:
+
+- %FLEET_URL%: Fleet Server URL to enroll the Elastic Agent into. FLEET_URL can be found in Kibana, go to Management > Fleet > Settings
+- %ENROLLMENT_TOKEN%: Elasticsearch API key used to [enroll Elastic Agents](https://www.elastic.co/guide/en/fleet/current/fleet-enrollment-tokens.html#fleet-enrollment-tokens) in Fleet. *This should be encoded as base64 value because it will be stored as Kubernetes secret*
+
+Eg.
+
+```yaml
+secretGenerator:
+  - name: elastic-agent-creds
+    literals:
+      - enrollment_token=%ENROLLMENT_TOKEN%
+```
+
+Comm
+
+For *Standalone Elastic Agent*, please update the following secrets inside main [kustomization.yaml](./elastic-agent-managed/kustomization.yaml):
+
+- %ES_HOST%: The Elasticsearch host to communicate with
+- %API_KEY: The API Key with access privilleges to connect to Elasticsearch. See [create-api-key-standalone-agent](https://www.elastic.co/guide/en/fleet/current/grant-access-to-elasticsearch.html#create-api-key-standalone-agent). *This should be encoded as base64 value because it will be stored as Kubernetes secret*
+- %CA_TRUSTED%: The ssl.ca_trusted_fingerprint in order the elastic agent to be able to trust specific certificate
+- %ELASTIC_AGENT_ID%: A numeric ELASTIC_AGENT_ID that will be added as a new field. *By default, this will be added to state_pod dataset.*
```

Comment:
Not sure if this is the right name for this variable, real Elastic Agent ID is not known at the time we run `kubectl ...`, something like `DATASET_ID` might work better.

Reply:
To be accurate this is the session id that will be created from kibana and will denote the specific agent or in general the installation process. Yes I dont mind to change it to whatever UI team things is more appopriate

Reply:
Awesome, thank you! `SESSION_ID` sounds to me a bit ephemeral, considering that it will be present for the whole lifespan of the agent config. `DATASET_ID` is the best I can think of now…

Oh, and btw, could we make it a string type so it supports UUID values?

Reply:
Thanks done: [111d8d0](https://github.com/elastic/elastic-agent/pull/4754/commits/111d8d0a3cb47e2941faf049e8eb3c4f20051bbf)